### ISSUE TITLE
AIX build failures looks like machine config

### ISSUE THREAD

**Issue by mhdawson** [nodejs/build#504]
For some reason builds on AIX seem to be failing with:  

/origin/* +refs/pull/31/head:refs/remotes/origin/_jenkins_local_branch # timeout=20
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from git@github.com:nodejs/node-private.git
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:799)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1055)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1086)
	at hudson.scm.SCM.checkout(SCM.java:485)
	at hudson.model.AbstractProject.checkout(AbstractProject.java:1269)
	at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:607)
	at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
	at hudson.model.Run.execute(Run.java:1738)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: hudson.plugins.git.GitException: Command "git -c core.askpass=true fetch --tags --progress git@github.com:nodejs/node-private.git +refs/heads/*:refs/remotes/origin/* +refs/pull/31/head:refs/remotes/origin/_jenkins_local_branch" returned status code 128:
stdout: 
stderr: OpenSSL version mismatch. Built against 1000105f, you have 100000bf
fatal: Could not read from remote repository.

**Comment by mhdawson** [comment]
Kicked off 2 runs to make sure it still recreates against head.

**Comment by mhdawson** [comment]
Seems to have gotten further on test-osuosl-aix61-ppc64_be-1

**Comment by mhdawson** [comment]
Got further on  test-osuosl-aix61-ppc64_be-2 as well.

**Comment by mhdawson** [comment]
Runs on both machines seems to be ok

**Comment by mhdawson** [comment]
Launched build to match the one that was failing.  It also failed.  They key difference seems to be that the GIT_ORIGIN_SCHEME as git versus https.  That does jive with the failure reported.

**Comment by mhdawson** [comment]
Local checkout with git seemed out.  Looking at config for jenkins agent it looks like this:

export LIBPATH=/opt/freeware/lib:$LIBPATH;

is what causes the problem.

**Comment by mhdawson** [comment]
Ok as a summary the issue was due to dependencies between openssl/openssh/git.  

1) We had installed git from bull freeware and it required export LIBPATH=/opt/freeware/lib to find the version of the libcrypto.a library that it needed.

2) ssh which is only used when we build from the private repo, however, however, could not run with the version of libcrypto.a in /opt/freeware/lib.  

Jenkins however uses both 1) and 2) when building from the private repo so there was no good way to make them both happy (despite my looking).

The resolution was to drop the bull freeware of git, install a version directly from IBM and to also update a number of dependencies to the latest so that all the pieces work together.  

I've Updated the PR which covers the AIX setup setups to reflect these changes.

I've also update both test machines and the release machine and seen passes of both types (public and private repo) on the test machines as well as the release being succesfully built so I'm going to close this issue.
